### PR TITLE
raspberry pi packages: pass through version info for auto-updates, first batch of auto-updates

### PR DESCRIPTION
**Pull request by colemickens**
###### Description of changes

cc: @lovesegfault

The first commit moves the rev/hashes into a `verinfo` attrset and then passes it thru meta, for the various raspberrypi firmware-esque packages.

I have a script that is a bit too messy to upstream right now, that knows how to update these packages according to what "upstream" recommends when you sufficiently read between the lines.

It's smart enough to update the Wifi+BT package and pick the newer version as the updated package version. It's smart enough to match the kernel commit to the version that is indicated in rpi eeprom repository, etc.

The subsequent commits are just the result of running my script. The script is flexible enough that it's fairly idempotent. I point it at some branch of nixpkgs (that has the `verinfo` commit), and a "target" branch, and it will reset the  target to the source, and then update all packages. (Meaning in theory, this script could run in a loop, auto-rebasing on itself and pushing, and it would just keep adding new commits as other packages get updated).

I am working on automated testing for this as well, but I need to get some of this rolling.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/firmware/raspberrypi/default.nix [hunk 1]
```diff
@@ -1,15 +1,21 @@
 { lib, stdenvNoCC, fetchFromGitHub }:
 
+let verinfo = {
+  # version = "2022-01-06";
```
Comment (colemickens) [inline pkgs/os-specific/linux/firmware/raspberrypi/default.nix:4]: note: I took the easy way out here and changed the external package version. Not sure this was the right call.

File: pkgs/os-specific/linux/raspberrypi-eeprom/default.nix [hunk 2]
```diff
@@ -1,15 +1,21 @@
 { stdenvNoCC, lib, fetchFromGitHub, makeWrapper
 , python3, binutils-unwrapped, findutils, kmod, pciutils, libraspberrypi
 }:
+
+let verinfo = {
+  version = "2022-04-26";
```
Comment (colemickens) [inline pkgs/os-specific/linux/raspberrypi-eeprom/default.nix:6]: unstable label dropped intentionally. They don't tag the repo reliably, the stable builds are just published into the stable/ directory which is what my script picks out of.

**Comment by colemickens** [comment]
this is the copy of the script I ran: https://github.com/colemickens/nixcfg/blob/0861ec882ee8e88121977f3e8389465c758c28b8/misc/rpi/update-rpi-packages.sh

**Review by lovesegfault — APPROVED** [review]

**Comment by RaitoBezarius** [comment]
Is this PR still relevant, can we do something to help this getting merged? cc @samueldr which might be your area of expertise.

**Comment by colemickens** [comment]
abandoning, I've given on SBCs.

**Pull request closed without merging**